src/uvutil.cpp

```diff
@@ -93,15 +130,41 @@ void ExtendedWrite::next() {
     return;
   }
   if (buf.len == 0) {
-    _pDataSource->freeData(buf);
-    if (_activeWrites == 0) {
-      _pDataSource->close();
-      onWriteComplete(0);
-    }
-    return;
+    // No more data is going to come.
+    // Ensure future calls to next() results in disposal (assuming that all
+    // outstanding writes are done).
+    _completed = true;
   }
+
   WriteOp* pWriteOp = new WriteOp(this, buf);
-  uv_write(&pWriteOp->handle, _pHandle, &pWriteOp->buffer, 1, &writecb);
+  if (this->_chunked) {
+    if (buf.len == 0) {
+      // In chunked mode, the last chunk must be followed by one more "\r\n".
+      pWriteOp->setSuffix(TRAILER.c_str(), TRAILER.length());
+    } else {
+      // In chunked mode, data chunks must be preceded by 1) the number of bytes
+      // in the chunk, as a hexadecimal string; and 2) "\r\n"; and succeeded by
+      // another "\r\n"
+      char prefix[16];
+      memset(prefix, 0, sizeof(prefix));
```

Comment:
I _think_ this should work, and seems safer and cleaner than using `memset()`.

```suggestion
      char prefix[16] = {0};
```